pkg/ibmvpc/connectivity_result.go

```diff
@@ -21,3 +26,96 @@ type ConnectivityResult struct {
 	deniedConns map[*ipblock.IPBlock]*connection.Set // denied target and its allowed connections, by deny rules.
 	denyRules   map[*ipblock.IPBlock][]int           // indexes of deny rules relevant to this connectivity
 }
+
+func equalConns(conns1, conns2 map[*ipblock.IPBlock]*connection.Set) bool {
+	if len(conns1) != len(conns2) {
+		return false
+	}
+	keys1 := make([]string, 0, len(conns1))
+	for i := range conns1 {
+		keys1 = append(keys1, i.String())
+	}
+	sort.Strings(keys1)
+	keys2 := make([]string, 0, len(conns2))
+	for i := range conns2 {
+		keys2 = append(keys2, i.String())
+	}
+	sort.Strings(keys2)
+	// compare the concatenation result to validate equality of keys sets
+	for i := 0; i < len(keys1); i++ {
+		if keys1[i] != keys2[i] {
+			return false
+		}
+	}
+	for ip, conn := range conns1 {
+		for otherIP, otherConn := range conns2 {
+			if ip.Equal(otherIP) {
+				if !conn.Equal(otherConn) {
+					return false
+				}
+				break
+			}
+		}
+	}
+	return true
+}
+
+func equalRules(rules1, rules2 map[*ipblock.IPBlock][]int) bool {
+	if len(rules1) != len(rules2) {
+		return false
+	}
+	keys1 := make([]string, 0, len(rules1))
+	for i := range rules1 {
+		keys1 = append(keys1, i.String())
+	}
+	sort.Strings(keys1)
+	keys2 := make([]string, 0, len(rules2))
+	for i := range rules2 {
+		keys2 = append(keys2, i.String())
+	}
+	sort.Strings(keys2)
+	// compare the concatenation result to validate equality of keys sets
+	for i := 0; i < len(keys1); i++ {
+		if keys1[i] != keys2[i] {
+			return false
+		}
+	}
+	for ip, indexes := range rules1 {
+		for otherIP, otherIndexes := range rules2 {
+			if ip.Equal(otherIP) {
+				if !reflect.DeepEqual(indexes, otherIndexes) {
```

Comment:
are the indexes guaranteed to be in the same order ? or need to be sorted?

Reply:
will sort before checking equality